osbs/utils.py

```diff
@@ -289,6 +289,10 @@ def get_time_from_rfc3339(rfc3339):
         return timegm(time_tuple)
 
 
+
+VALID_BUILD_CONFIG_NAME_CHARS = re.compile('[a-z0-9]')
```

Comment:
Oh, striping actually should happen after the length of the members was adjusted, something like:
```python
VALID_BUILD_CONFIG_NAME_CHARS = re.compile('[\-a-z0-9]')
MEMBER_LIMIT = (limit - hash_size - 2) // 2

# ... calculate hash

repo = filter(VALID_BUILD_CONFIG_NAME_CHARS.match, list(repo))
repo = ''.join(repo)
repo = repo[:min(MEMBER_LIMIT, len(repo))]
repo = repo.strip('-')

branch = filter(VALID_BUILD_CONFIG_NAME_CHARS.match, list(branch))
branch = ''.join(branch)
branch = branch[:min(MEMBER_LIMIT, len(branch))]
branch = branch.strip('-')
```